pydantic/_internal/_validate_call.py

```diff
@@ -97,3 +102,77 @@ def __call__(self, *args: Any, **kwargs: Any) -> Any:
         if self.__return_pydantic_validator__:
             return self.__return_pydantic_validator__(res)
         return res
+
+
+class ValidateCallInfo(TypedDict):
+    validate_return: bool
+    config: ConfigDict | None
+    function: Callable[..., Any]
+    local_namspace: dict[str, Any] | None
+
+
+def _is_wrapped_by_validate_call(obj: object) -> bool:
+    return hasattr(obj, '__pydantic_validate_call_info__')
+
+
+def collect_validate_call_info(namespace: dict[str, Any]):
+    validate_call_infos = {}
+    for name, func in namespace.items():
+        if _is_wrapped_by_validate_call(func):
+            validate_call_infos[name] = func.__pydantic_validate_call_info__
+
+    return validate_call_infos
+
+
+def _replicate_validate_call(info: ValidateCallInfo) -> Callable[..., Any]:
+    """When normally calling `validate_call`, we use the namespace of the frame that called it as local_ns.
+    This function mock that behavior by calling `validate_call` inside a new frame where we have copied all
+    local variables into.
```

Comment:
I'm not sure I follow what the docstring is saying, where is this copy happening?

Reply:
This is outdated. I will remove it.